From 8059bfb84275776b402534dcfb6a01ca2984d98b Mon Sep 17 00:00:00 2001
From: yingzhex <yingzhenx.li@intel.com>
Date: Thu, 17 Jun 2021 15:41:39 +0800
Subject: [PATCH] Add
 talk/owt/patches/0019-Support-h264-and-hevc-of-MTK-device-for-application.patch

Signed-off-by: yingzhex <yingzhenx.li@intel.com>
---
 scripts/prepare_dev.py                        |  3 +-
 ...d-hevc-of-MTK-device-for-application.patch | 54 +++++++++++++++++++
 2 files changed, 56 insertions(+), 1 deletion(-)
 create mode 100644 talk/owt/patches/0019-Support-h264-and-hevc-of-MTK-device-for-application.patch

diff --git a/scripts/prepare_dev.py b/scripts/prepare_dev.py
index f3de997..b4450c2 100644
--- a/scripts/prepare_dev.py
+++ b/scripts/prepare_dev.py
@@ -46,7 +46,8 @@ patchList = [
     ('0015-Adopt-SingletonSurfaceView.patch', WEBRTC_PATH),
     ('0016-Use-AToU-to-print-trace.patch', WEBRTC_PATH),
     ('0017-Support-OMX.Intel.hw_ve.h264-for-android-app.patch', WEBRTC_PATH),
-    ('0018-Support-OMX.Intel.hw_ve.h265-for-android-app.patch', WEBRTC_PATH)
+    ('0018-Support-OMX.Intel.hw_ve.h265-for-android-app.patch', WEBRTC_PATH),
+    ('0019-Support-h264-and-hevc-of-MTK-device-for-application.patch', WEBRTC_PATH)
 ]
 
 def _patch(ignoreFailures=False):
diff --git a/talk/owt/patches/0019-Support-h264-and-hevc-of-MTK-device-for-application.patch b/talk/owt/patches/0019-Support-h264-and-hevc-of-MTK-device-for-application.patch
new file mode 100644
index 0000000..5c44eb3
--- /dev/null
+++ b/talk/owt/patches/0019-Support-h264-and-hevc-of-MTK-device-for-application.patch
@@ -0,0 +1,54 @@
+From 10630877cbe10cd02a9f99b70c136bba84e8c48e Mon Sep 17 00:00:00 2001
+From: yingzhex <yingzhenx.li@intel.com>
+Date: Thu, 17 Jun 2021 15:36:40 +0800
+Subject: [PATCH] Support h264 and hevc of MTK device for application
+
+Signed-off-by: yingzhex <yingzhenx.li@intel.com>
+---
+ sdk/android/api/org/webrtc/HardwareVideoEncoderFactory.java | 3 +++
+ sdk/android/src/java/org/webrtc/MediaCodecUtils.java        | 1 +
+ 2 files changed, 4 insertions(+)
+
+diff --git a/sdk/android/api/org/webrtc/HardwareVideoEncoderFactory.java b/sdk/android/api/org/webrtc/HardwareVideoEncoderFactory.java
+index f979d70487..2c8ac87ca2 100644
+--- a/sdk/android/api/org/webrtc/HardwareVideoEncoderFactory.java
++++ b/sdk/android/api/org/webrtc/HardwareVideoEncoderFactory.java
+@@ -15,6 +15,7 @@ import static org.webrtc.MediaCodecUtils.INTEL_PREFIX;
+ import static org.webrtc.MediaCodecUtils.QCOM_PREFIX;
+ import static org.webrtc.MediaCodecUtils.HISI_PREFIX;
+ import static org.webrtc.MediaCodecUtils.IMG_PREFIX;
++import static org.webrtc.MediaCodecUtils.MTK_PREFIX;
+ 
+ import android.media.MediaCodecInfo;
+ import android.media.MediaCodecList;
+@@ -252,6 +253,7 @@ public class HardwareVideoEncoderFactory implements VideoEncoderFactory {
+         || (name.startsWith(HISI_PREFIX) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT)
+         || (name.startsWith(IMG_PREFIX) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT)
+         || (name.startsWith(INTEL_PREFIX) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT)
++        || (name.startsWith(MTK_PREFIX) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT)
+         || vcp.isExtraHardwareSupported(name, "video/avc", vcp.parseWithTag(vcp.loadWithDom(extraMediaCodecFile), "Encoders"));
+   }
+ 
+@@ -266,6 +268,7 @@ public class HardwareVideoEncoderFactory implements VideoEncoderFactory {
+            || (name.startsWith(HISI_PREFIX) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT)
+            || (name.startsWith(IMG_PREFIX) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT)
+            || (name.startsWith(INTEL_PREFIX) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT)
++           || (name.startsWith(MTK_PREFIX) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT)
+            || vcp.isExtraHardwareSupported(name, "video/hevc", vcp.parseWithTag(vcp.loadWithDom(extraMediaCodecFile), "Encoders"));
+   }
+ 
+diff --git a/sdk/android/src/java/org/webrtc/MediaCodecUtils.java b/sdk/android/src/java/org/webrtc/MediaCodecUtils.java
+index 03b08a0878..b9636d58e4 100644
+--- a/sdk/android/src/java/org/webrtc/MediaCodecUtils.java
++++ b/sdk/android/src/java/org/webrtc/MediaCodecUtils.java
+@@ -29,6 +29,7 @@ class MediaCodecUtils {
+   static final String NVIDIA_PREFIX = "OMX.Nvidia.";
+   static final String QCOM_PREFIX = "OMX.qcom.";
+   static final String HISI_PREFIX = "OMX.hisi.";
++  static final String MTK_PREFIX = "OMX.MTK.";
+   static final String IMG_PREFIX = "OMX.IMG.";
+   static final String[] SOFTWARE_IMPLEMENTATION_PREFIXES = {"OMX.google.", "OMX.SEC."};
+ 
+-- 
+2.17.1
+
-- 
2.17.1

